assembly/src/ast/procedure/name.rs

```diff
@@ -349,7 +347,7 @@ impl FromStr for ProcedureName {
             Some((_, c)) if c.is_ascii_uppercase() => Err(IdentError::Casing(CaseKindError::Snake)),
             Some(_) => Err(IdentError::InvalidChars { ident: s.into() }),
         }?;
-        Ok(Self(Ident::new_unchecked(Span::unknown(raw))))
+        Ok(Self(Ident::from_raw_parts(Span::unknown(raw))))
```

Comment:
Not from this PR, but I found the code above a bit difficult to read. It might make sense to refactor it a bit to improve readability or maybe just add comments specifying what each branch is for. My understanding is:

- The second branch handle quoted strings.
- The third branch handles non-quoted strings, but makes sure that the first character is a lower-case character or one of the allowed symbols.

Reply:
Probably the best thing for readability here would be to add some comments describing the purpose of each part of the code, as you suggested.

We could also split the implementation into two functions, one for each identifier type, rather than handle those in two separate cases of a `match`, but not sure if that helps or not.

Reply:
Yes - I think adding some comments is probably sufficient.